### ISSUE TITLE
Support GitHub Copilot

### ISSUE THREAD

**Issue by modz2014** [notepad-plus-plus/notepad-plus-plus#11038]
hi i was wondering if you the feature for automatically typing in python scripts scripts and also the part that visual studio does when when it already knows what your going to type in the line 

i was on a website yesterday i cant of the add but it fixes missing in the line of code when writting in python if i find ill let you know

**Comment by Ekopalypse** [comment]
Please explain in more detail what you mean, currently there are several possible interptetations, such as copilot function, language server implementation or basic completions or ... of your question possible.

**Comment by modz2014** [comment]
well viula studio if your writting in C# it already knows what you need to write example button12.button = new System.Windows.Forms(); as soon as type = already knows the what i want to write

**Comment by Ekopalypse** [comment]
There is a reason why there are specific IDEs for a specific programming language.
To make this possible for a general-purpose text editor, you have to sponsor the data centre, which is then used to collect millions of lines of source code from different programming languages and implement AI models to train and respond to them. It is unlikely that this will happen soon.

**Comment by modz2014** [comment]
ok if you have discord i can chat with you about getting this done if your interested im pretty keen on getting this done same for python as well

**Comment by Ekopalypse** [comment]
Sorry, you got the wrong impression, I am not a Notepad++ developer as I have a weak understanding of cpp. Nevertheless, I would be very interested to know how you want to realise this with Python for example?

**Comment by modz2014** [comment]
im thinking of adding this notepad++ makes it easier for myslef and downloading visual studio all the time to fix mistake is annoying but anyway i live notepad++ more then any other app

**Comment by modz2014** [comment]
when is github-copilot released do you know hoping to use this in notepad++

**Comment by Ekopalypse** [comment]
No, LOL - I doubt Microsoft will make this available for free to non-MS editors.
Btw. you didn't really answer my question HOW you would solve such a completion idea for e.g. Python.

**Comment by modz2014** [comment]
what do you i realise this in C++ im not very good with this

**Comment by ArkadiuszMichalski** [comment]
After buying GitHub, they have a nice codebase for analysis ([info](https://en.wikipedia.org/wiki/OpenAI), [info).](https://en.wikipedia.org/wiki/GitHub_Copilot) 
>OpenAI Codex is a descendant of GPT-3 that has additionally been trained on code from 54 million GitHub repositories.[82][83] It was announced in mid-2021 as the AI powering the code autocompletion tool GitHub Copilot.

**Comment by mere-human** [comment]
Did you mean Intellisense or autocompletion?
![image](https://user-images.githubusercontent.com/9664141/149504702-2b10e78c-789d-4bad-9599-750f614fd3bf.png)

**Comment by modz2014** [comment]
well im hoping to creating a plugin for github copilt for notepad++ but i dont have access to it atm

**Comment by modz2014** [comment]
> No, LOL - I doubt Microsoft will make this available for free to non-MS editors. Btw. you didn't really answer my question HOW you would solve such a completion idea for e.g. Python.

do you have discord

**Comment by mlisowsk** [comment]
I am seconding the feature request. Not just for phyton but for Javascript.
Specifically I desire the ability of Github Copilot to provide "suggestions for whole lines or entire functions right inside your editor." See https://copilot.github.com/
Neovim is supports Github Copilot, so I don't see a reason why NPP shouldn't technically be able to use Github Copilot as well.

**Comment by Ekopalypse** [comment]
As far as I know, there is no official publicly available API/service to connect to.
To quote from the linked site: `We currently support Visual Studio Code, Neovim, and IntelliJ-based IDEs like JetBrains IntelliJ IDEA, Android Studio or PyCharm.`
If this is not the case, a link to the API would be helpful.

**Comment by modz2014** [comment]
Ok dump the api haha

**Comment by modz2014** [comment]
@donho hey i have question about the shell extension when I was looking at the msix package in the window apps folder looking at the XML code `<com:Class Id="E6950302-61F0-4FEB-97DB-855E30D4A991" Path="NppShell.dll" ThreadingModel="STA"/>` how does this know where the .dll is located
